weave/trace/serialization/serializer.py

```diff
@@ -36,14 +36,54 @@ def load_instance(
 
 from __future__ import annotations
 
+import inspect
 from dataclasses import dataclass
-from typing import Any, Callable
+from typing import Any, Callable, Union
+
+from typing_extensions import TypeGuard
+
+# Not locking down the return type for inline encoding but
+# it would be expected to be something like a str or dict.
+InlineSave = Callable[[Any], Any]
+# Second parameter should really be MemTraceFilesArtifact
+# but we are avoiding a circular import.
+FileSave = Callable[[Any, Any, str], None]
+Save = Union[InlineSave, FileSave]
```

Comment:
And with future you can do `InlineSave | FileSave` heree

Reply:
I tried that but even with with future it fails on 3.9:
```
TypeError: unsupported operand type(s) for |: '_CallableGenericAlias' and '_CallableGenericAlias'
```

Reply:
ok no worries